src/Microsoft.AspNet.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -23,23 +24,42 @@ public static IApplicationBuilder UseMiddleware(this IApplicationBuilder builder
             return builder.Use(next =>
             {
                 var instance = ActivatorUtilities.CreateInstance(builder.ApplicationServices, middleware, new[] { next }.Concat(args).ToArray());
-                var methodinfo = middleware.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public);
+                var methods = middleware.GetMethods(BindingFlags.Instance | BindingFlags.Public);
```

Comment:
Do you want to do these checks before you try to create the instance? There's no point in creating the instance if any of these conditions fail